serverless/client/client.go

```diff
@@ -48,6 +48,19 @@ import (
 // based implementation MUST return this error when it receives a 404 StatusCode.
 type Fetcher func(ctx context.Context, path string) ([]byte, error)
 
+// ConsensusCheckpoint is a function which returns the largest checkpoint known which is
+// signed by logSigV and satisfies some consensus algorithm.
+//
+// This is intended to provide a hook for adding a consensus view of a log, e.g. via witnessing.
+type ConsensusCheckpoint func(ctx context.Context, logSigV note.Verifier, origin string) (*log.Checkpoint, []byte, error)
```

Comment:
`ConsensusCheckpoint` sounds like a noun, but the type is a function. Maybe `CheckpointConsensusFn` ?

Reply:
+1 to this!

Reply:
👍  done